### DISCUSSION TITLE
[BP#11] Utiliser un framework ou développer sur mesure

### DISCUSSION THREAD

**Discussion by devcrafting** [cnumr/best-practices#77] [Bonnes pratiques] [answered]
Le titre laisse penser à "qqch de grand" et le contenu me semble très restrictif. Je suis parfaitement en phase avec le fait qu'il faille se poser la question du choix d'un framework VS du sur-mesure mais l'exemple me paraît très dans le détail.
Je parlerais plus des soucis des framework (en citant d'autres exemples qui parlent à différentes populations : ORM, framework Front, en plus d'un CMS comme Drupal). Ces soucis étant principalement d'embarquer beaucoup de fonctionnalités non utilisées, d'apporter des contraintes qui peuvent aller contre la sobriété numérique par un fonctionnement sous-optimal (cf. les autres pratiques). On peut certainement donner des exemples.

**Comment by tbroyer** [comment, marked as answer]
La BP n'est pas "framework vs sur-mesure", mais "CMS vs sur-mesure" (pourquoi framework est utilisé ici ? je ne sais pas, si on utilise un framework, on fera forcément du sur-mesure avec…)
Je serais par contre plutôt d'accord d'opposer aussi framework vs no-framework.

Et oui, effectivement l'exemple cité est très réducteur !

Je me demande d'ailleurs dans quelle mesure cette BP n'entre pas en partie en conflit avec la BP#14, où une architecture modulaire incite à utiliser les hooks d'extension décriés ici.

**Reply by devcrafting** [reply]
je n'avais pas compris en "CMS vs sur-mesure". 

oui je suis assez d'accord que c'est un peu en conflit avec la BP#14

à mon sens on peut reprendre le sens du titre. NB: utiliser un framework induit une sorte de sur-mesure par rapport à l'utilisation d'un CMS (ou autre "progiciel"), cependant on peut aussi considérer un choix entre utiliser un framework et faire du sur-mesure (exemple en CSS discuté dans d'autres BP ou avec la notion d'ORM pour l'accès aux données...)

**Reply by ACTLEM** [reply]
J'ai créé l'issue https://github.com/cnumr/best-practices/issues/327 pour reformuler la BP.

**Comment by florinesueur** [comment]
Il me semble que c'est presque qu'un redit de la [BP#015](https://github.com/cnumr/best-practices/blob/main/chapters/BP_015_fr.md) "Choisir les technologies les plus adaptées". Il me semble que le tout peut-être regroupable.

**Reply by devcrafting** [reply]
Du coup par rapport à ma proposition de découper la BP 15, celle-ci pourrait être celle concernant les solutions sur-mesure avec/sans framework

**Comment by devcrafting** [comment]
On fait quoi de cette BP? Il me semble qu'il y a du boulot pour la rendre plus claire, et en effet la BP#15 est très proche => à fusionner comme proposé par @florinesueur (plutôt dans la 15)

J'ouvre une discussion sur la 15 : https://github.com/cnumr/best-practices/discussions/199

**Reply by ACTLEM** [reply]
Si on choisit de supprimer la BP15 pour la spliter en plusieurs plus concrètes, cette BP garde tout son sens, non?

**Reply by ACTLEM** [reply]
En attente décision https://github.com/cnumr/best-practices/discussions/199

**Comment by Fred-artwai** [comment]
Je vote pour la suppression de cette BP.
